charts/redpanda/testdata/template-cases.txtar

```diff
@@ -948,3 +948,174 @@ statefulset:
+
+-- trust-store-reference --
+# ASSERT-NO-ERROR
+# ASSERT-GOLDEN
+auth:
+  sasl:
+    enabled: false
+
+external:
+  enabled: true
+  type: LoadBalancer
+  domain: <our domain>
+  addresses:
+    - redpanda-0
+    - redpanda-1
+    - redpanda-2
+  externalDns:
+    enabled: true
+  service:
+    enabled: true
+
+tls:
+  enabled: true
+  certs:
+    default:
+      caEnabled: true
+      secretRef:
+        name: redpanda-tls-cert
+      clientSecretRef:
+        name: redpanda-admin-cert
+
+listeners:
+  admin:
+    port: 9643
+    tls:
+      enabled: true
+      cert: default
+      requireClientAuth: true
+      trustStore:
+        configMapKeyRef:
+          name: redpanda-company-cacrt
+          key: ca.crt
+    external:
+      default:
+        port: 9644
+        tls:
+          enabled: true
+          cert: default
+          requireClientAuth: true
+          trustStore:
+            configMapKeyRef:
+              name: redpanda-company-cacrt
+              key: ca.crt
+        advertisedPorts:
+          - 9644
+  kafka:
+    port: 9093
+    tls:
+      enabled: true
+      cert: default
+      requireClientAuth: true
+      trustStore:
+        configMapKeyRef:
+          name: redpanda-company-cacrt
+          key: ca.crt
+    external:
+      default:
+        port: 9094
+        tls:
+          enabled: true
+          cert: default
+          requireClientAuth: true
+          trustStore:
+            configMapKeyRef:
+              name: redpanda-company-cacrt
+              key: ca.crt
+        advertisedPorts:
+          - 9094
+  http:
+    enabled: true
+    port: 8082
+    kafkaEndpoint: default
+    tls:
+      enabled: true
+      cert: default
+      requireClientAuth: true
+      trustStore:
+        configMapKeyRef:
+          name: redpanda-company-cacrt
+          key: ca.crt
+    external:
+      default:
+        port: 8083
+        tls:
+          enabled: true
+          cert: default
+          requireClientAuth: true
+          trustStore:
+            configMapKeyRef:
+              name: redpanda-company-cacrt
+              key: ca.crt
+        advertisedPorts:
+          - 8083
+  rpc:
+    port: 33145
+    tls:
+      enabled: true
+      cert: default
+      requireClientAuth: true
+      trustStore:
+        configMapKeyRef:
+          name: redpanda-company-cacrt
+          key: ca.crt
+  schemaRegistry:
+    enabled: true
+    port: 8081
+    kafkaEndpoint: default
+    tls:
+      enabled: true
+      cert: default
+      requireClientAuth: true
+      trustStore:
+        configMapKeyRef:
+          name: redpanda-company-cacrt
+          key: ca.crt
+    external:
+      default:
+        port: 8084
+        tls:
+          enabled: true
+          cert: default
+          requireClientAuth: true
+          trustStore:
+            configMapKeyRef:
+              name: redpanda-company-cacrt
+              key: ca.crt
+        advertisedPorts:
+          - 8084
+
+console:
```

Comment:
Is console relevant to this test case? If so, let's add in an assertion. If not, could we remove it to further scope this test case down to the issue at hand?

Reply:
I will remove.